ngraph/core/src/op/interpolate.cpp

```diff
@@ -239,11 +239,20 @@ void op::v4::Interpolate::validate_and_infer_types()
         return;
     }
 
+    const auto input_rank = input_shape.rank().get_length();
+
+    // If the input 'axes' is given and this input is not Constant, we cannot infer any elements
+    // of the output shape. Hence, all components of the output shape should be dynamic.
+    if (input_values().size() == 4 && !is_type<op::Constant>(input_value(3).get_node()))
+    {
+        PartialShape output_shape = std::vector<Dimension>(input_rank, Dimension::dynamic());
+        set_output_type(0, get_input_element_type(0), output_shape);
+        return;
+    }
+
     auto axes = get_axes();
     correct_pads();
 
-    const auto input_rank = input_shape.rank().get_length();
-
     PartialShape padded_input_shape = get_padded_input_shape(input_shape);
     PartialShape output_shape = padded_input_shape;
 
```

Comment:
Is interval logic used for calculation `output_shape`? I think `infer_using_scales` and `infer_using_shapes` should be updated with it.

Reply:
Interval logic is applicable only when `shape_calculation_mode == 'scales'`, the input `scales` is constant, and the input `axes` is constant. Implementation of the interval logic (for this case) should be done in the separate PR, after merge this PR.